client/src/components/Review Widget/ReviewMetaData.jsx

```diff
@@ -56,7 +50,19 @@ const ReviewMetaData = (props) => {
     }
 
     currentAverage = Math.round((currentAverage / reviewCount) * 10) / 10
-    calculatePercentRecommend(currentAverage, data)
+    calculatePercentRecommend(currentAverage, data, reviewCount)
+  }
+
+  const calculatePercentRecommend = (currentAverage, data, reviewCount) => {
+    const noCount = parseInt(data.recommended.false)
+    const yesCount = parseInt(data.recommended.true)
+
+    setReviewsMeta({
+      reviewsMeta: data,
+      average: currentAverage,
+      percentRecommend: Math.round((yesCount / (noCount + yesCount === 0 ? 1 : (noCount + yesCount))) * 100),
+      totalReviews: reviewCount
+    })
   }
```

Comment:
It looks like there is a bit of coupling going on between these functions. I can see the logic path here but I would advise decoupling them and instead move the chaining logic to the original invocation. This way, they stay helper functions. 

My recommendation here, if a function is named a specific way, it should return what it is named. 
This also goes for the get request, just as an example, you could have it return a promise instead of calling another function that then calls another function that then writes to state.

Reply:
Awesome math skills here. I agree with the decoupling. Perhaps something like percentRecommend: getPercent() instead of percentRecommend: Math.round((yesCount / (noCount + yesCount === 0 ? 1 : (noCount + yesCount))) * 100) (or doing calculations before the assignment and setting it there).